src/main/java/com/faforever/client/replay/LiveReplayService.java

```diff
@@ -0,0 +1,124 @@
+import javafx.util.Pair;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.util.concurrent.Future;
+
+import static com.faforever.client.util.Assert.checkNullIllegalState;
+
+@Lazy
+@Service
+@RequiredArgsConstructor
+public class LiveReplayService implements InitializingBean, DisposableBean {
+
+  private final ClientProperties clientProperties;
+  private final TaskScheduler taskScheduler;
+  private final NotificationService notificationService;
+  private final I18n i18n;
+  private final ReplayService replayService;
+  private final GameService gameService;
+
+  private Integer watchDelaySeconds;
+  private Future<?> futureTask;
+  private final ObjectProperty<Pair<Integer, LiveReplayAction>> trackingReplayProperty = new SimpleObjectProperty<>(null);
+
+  @Override
+  public void afterPropertiesSet() throws Exception {
+    watchDelaySeconds = clientProperties.getReplay().getWatchDelaySeconds();
+    JavaFxUtil.addListener(gameService.gameRunningProperty(), (observable, oldValue, newValue) -> {
+      if (newValue.equals(true) && trackingReplayProperty.getValue() != null) {
+        stopTrackingReplay();
+      }
+    });
+  }
+
+  public boolean canWatchReplay(GameBean game) {
+    Duration duration = getWatchDelayTime(game);
+    return duration.isZero() || duration.isNegative();
+  }
+
+  public Duration getWatchDelayTime(GameBean game) {
+    Assert.notNull(game.getStartTime(), "Game's start time is null, in which case it shouldn't even be listed: " + game);
+    return Duration.between(OffsetDateTime.now(), game.getStartTime().plusSeconds(watchDelaySeconds)
+    );
+  }
+
+  public void performActionWhenAvailable(GameBean game, LiveReplayAction action) {
+    checkNullIllegalState(game.getId(), "No game id to schedule future task");
+    stopTrackingReplay();
+    trackingReplayProperty.set(new Pair<>(game.getId(), action));
+    switch (action) {
+      case NOTIFY_ME -> notifyUserWhenReplayIsAvailable(game);
+      case RUN -> runLiveReplayWhenIsAvailable(game);
+    }
+  }
+
+  private void notifyUserWhenReplayIsAvailable(GameBean game) {
+    futureTask = taskScheduler.schedule(() -> {
+      clearTrackingReplayProperty();
+      notificationService.addNotification(new TransientNotification(
+          i18n.get("vault.liveReplays.notifyMe.replayAvailable", game.getTitle()),
+          i18n.get("vault.liveReplays.notifyMe.replayAvailable.click"),
+          null,
+          (event) -> replayService.runLiveReplay(game.getId())));
+    }, Instant.from(game.getStartTime().plusSeconds(watchDelaySeconds)));
```

Comment:
I would probably make this a persistent notification that way if someone steps away they will still be able to interact with it when they get back.